model-optimizer/extensions/middle/MergeNodesPermutations.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from extensions.middle.ApplyNHWCtoNCHWpermutation import ApplyNHWCtoNCHWpermutation
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Graph, Node
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.utils.error import Error
+
+
+class MergeNodesPermutations(MiddleReplacementPattern):
+    enabled = True
+
+    def run_after(self):
+        return [ApplyNHWCtoNCHWpermutation]
+
+    def run_before(self):
+        return []
+
+    def find_and_replace_pattern(self, graph: Graph):
+        self.merge_nodes_permutations(graph)
+
+    @staticmethod
+    def merge_nodes_permutations(graph: Graph):
+        # Iterate over all data nodes and check all permutations for similarity
+        # In case of equal permutations, this permutation will be set as attribute for data node
+        # otherwise exception will be raised
+        for node in graph.nodes():
+            node = Node(graph, node)
+            if node.kind != 'data':
+                continue
+
+            permutations = []
+
+            # Get all permutations from in edges
+            for in_node in node.in_nodes():
+                edge_attrs = node.graph.get_edge_data(in_node.id, node.id)[0]
+                if 'permutation' in edge_attrs:
+                    permutations.append(edge_attrs['permutation'])
+
+            # Get all permutations from out edges
+            for out_node in node.out_nodes():
+                edge_attrs = node.graph.get_edge_data(node.id, out_node.id)[0]
+                if 'permutation' in edge_attrs:
+                    permutations.append(edge_attrs['permutation'])
+
+            # Check that all permutations are equal
```

Comment:
nit: in this line we are not checking but rather filling with default values. I propose move this comment into line 59 or even remove it since the following code is self-explanatory